src/feature/client/transports.c

```diff
@@ -1902,3 +1933,46 @@ managed_proxy_severity_parse(const char *severity)
 
   return -1;
 }
+
+/** Return the outbound address from the given <b>family</b>. Returns NULL if
+ * the user haven't specified a specific outbound address in either
+ * OutboundBindAddress or OutboundBindAddressPT. */
+STATIC const tor_addr_t *
+managed_proxy_outbound_address(const or_options_t *options, sa_family_t family)
+{
+  tor_assert(options);
+
+  const tor_addr_t *address = NULL;
+  int family_index;
+
+  switch (family) {
```

Comment:
There are now two copies of this code in tor, the other copy is in conn_get_outbound_address().
Let's refactor this code into a new function, maybe in config.c?